From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?ZX=E5=A4=8F=E5=A4=9C=E4=B9=8B=E9=A3=8E?=
 <snwcreations@qq.com>
Date: Thu, 17 Oct 2024 20:51:18 +0800
Subject: [PATCH] fix: Mohist compatibility

This commit is made to make the plugin be compatible with Mohist the server software.

diff --git a/dist/pom.xml b/dist/pom.xml
index 928f2607faf89922c686841a05a98264b30d6216..5b2c0958d29713805b78c0f217fc98a6a648a71a 100644
--- a/dist/pom.xml
+++ b/dist/pom.xml
@@ -181,7 +181,7 @@
                 <artifactId>maven-jar-plugin</artifactId>
                 <version>3.3.0</version>
                 <configuration>
-                    <finalName>Denizen-MC1.20.1-${DENIZEN_VERSION}-b${BUILD_NUMBER}-${BUILD_CLASS}</finalName> <!-- Backport - Mark the plugin as only supporting MC 1.20.1 -->
+                    <finalName>Denizen-MC1.20.1-${DENIZEN_VERSION}-b${BUILD_NUMBER}-${BUILD_CLASS}-Mohist</finalName> <!-- Backport - Mark the plugin as only supporting MC 1.20.1 --> <!-- Mohist Compatibility - The built plugin is only for Mohist -->
                     <outputDirectory>../target</outputDirectory>
                     <archive>
                         <manifest>
diff --git a/v1_20/pom.xml b/v1_20/pom.xml
index e3e74853ef94f989d41870b7704147adc87535f1..3c1d1212d3fc0384fb60812a02d9d41261a8e6a1 100644
--- a/v1_20/pom.xml
+++ b/v1_20/pom.xml
@@ -8,6 +8,15 @@
     <artifactId>denizen-v1_20</artifactId>
     <version>1.3.1-SNAPSHOT</version>
 
+    <!-- Mohist Compatibility start - Add Mohist to dependencies, so we can look at NMS things using SRG mapping -->
+    <repositories>
+        <repository>
+            <id>mohistmc</id>
+            <url>https://maven.mohistmc.com/</url>
+        </repository>
+    </repositories>
+    <!-- Mohist Compatibility end -->
+
     <dependencies>
         <dependency>
             <groupId>com.denizenscript</groupId>
@@ -16,19 +25,28 @@
             <scope>compile</scope>
             <optional>true</optional>
         </dependency>
+        <!-- Mohist Compatibility start - Add Mohist to dependencies so we can look at NMS things using SRG mapping
         <dependency>
             <groupId>org.spigotmc</groupId>
             <artifactId>spigot-api</artifactId>
-            <version>1.20.1-R0.1-SNAPSHOT</version> <!-- Backport - Use 1.20.1 for this module -->
+            <version>1.20.1-R0.1-SNAPSHOT</version>
             <scope>provided</scope>
         </dependency>
+        -->
         <dependency>
             <groupId>org.spigotmc</groupId>
             <artifactId>spigot</artifactId>
-            <version>1.20.1-R0.1-SNAPSHOT</version> <!-- Backport - Use 1.20.1 for this module -->
+            <version>1.20.1-R0.1-SNAPSHOT</version>
             <classifier>remapped-mojang</classifier>
             <scope>provided</scope>
         </dependency>
+        <dependency>
+            <groupId>com.mohistmc</groupId>
+            <artifactId>mohistdev</artifactId>
+            <version>1.20.1</version>
+            <scope>provided</scope>
+        </dependency>
+        <!-- Mohist Compatibility end -->
     </dependencies>
 
     <build>
diff --git a/v1_20/src/main/java/com/denizenscript/denizen/nms/v1_20/helpers/EntityHelperImpl.java b/v1_20/src/main/java/com/denizenscript/denizen/nms/v1_20/helpers/EntityHelperImpl.java
index cc7ca972fa16b6966c830ac267df142dba3571f8..ae42bef1b3025429b779fb8ec1844e38da03b809 100644
--- a/v1_20/src/main/java/com/denizenscript/denizen/nms/v1_20/helpers/EntityHelperImpl.java
+++ b/v1_20/src/main/java/com/denizenscript/denizen/nms/v1_20/helpers/EntityHelperImpl.java
@@ -758,7 +758,7 @@ public class EntityHelperImpl extends EntityHelper {
     }
 
     // Use reflection because Paper changes the method return type
-    public static final MethodHandle PLAYERLIST_REMOVE = ReflectionHelper.getMethodHandle(PlayerList.class, "remove", ServerPlayer.class);
+    public static final MethodHandle PLAYERLIST_REMOVE = ReflectionHelper.getMethodHandle(PlayerList.class, "m_11286_", ServerPlayer.class); // Mohist Compatibility - remove(ServerPlayer) method has not been remapped in Mohist mapping
 
     @Override
     public void setUUID(Entity entity, UUID id) {
